### ISSUE TITLE
Add Support for Enums in Config Beans

### ISSUE THREAD

**Issue by karthicks** [typesafehub/config#397]
Currently, a config bean cannot define a property whose type is an `Enum`, or a list of `Enum` for that matter. Ideally, one should be able to specify the value of the `Enum` as a `ConfigValue` whose type is `ConfigValueType.STRING`. Similarly, if the bean property is a list of `Enum`, one should be able to specify it's value as a `ConfigList` of such `ConfigValues`.

To illustrate this, consider the following bean, which contains two `Enum`-based properties:

```
 public class EnumsConfig {
    public enum Problem {
        P1, P2, P3;
    };
    public enum Solution {
        S1, S2, S3;
    }
    Problem problem;
    List<Solution> solutions;
```

One should be able to configure this object like so:

```
"enums" : {
    "problem" : "P1",
    "solutions" : ["S1", "S3"]
}
```

Karthick Sankarachary

**Comment by karthicks** [comment]
@havocp, can we close this issue, given that the pull request was merged, and is now part of v1.3.1?

**Comment by havocp** [comment]
Thanks, sounds good :-)
